crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -128,64 +135,58 @@ pub struct ModuleResolutionSettings {
     /// The root of the workspace, used for finding first-party modules.
     pub workspace_root: FileSystemPathBuf,
 
+    /// Optional (already validated) path to standard-library typeshed stubs.
+    /// If this is not provided, we will fallback to our vendored typeshed stubs
+    /// bundled as a zip file in the binary
+    pub custom_typeshed: Option<FileSystemPathBuf>,
```

Comment:
I wonder if we should change the settings to `Option<(FileSystemPathBuf, TypeshedVersions)>`. This way, parsing the typeshed versions happens outside of salsa (solves the error reporting problem).

Reply:
Hmm, but we need to re-parse the `VERSIONS` file every time it changes if we're in an LSP context

Reply:
But the need for recomputing also applies to the `custom_typeshed` path? 

Salsa will re-run all `resolve_module` queries when the `ModuleResolutionSettings` change. The only thing that happens outside (at least for now) is how we react to setting changes.

Reply:
The user won't pass a direct path to the VERSIONS file on the command line or in a config file. They'll pass a direct path to the custom typeshed directory, and the VERSIONS file is found inside that directory. If they then edit the contents of the VERSIONS file (or delete it entirely), we'll need to react to that change and attempt to re-parse the VERSIONS file in the custom typeshed directory, even though none of the settings passed by the user will have changed in any way